SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -240,6 +241,30 @@ Worker_EntityId USpatialPackageMapClient::GetEntityIdFromObject(const UObject* O
 	return GetUnrealObjectRefFromNetGUID(NetGUID).Entity;
 }
 
+AActor* USpatialPackageMapClient::GetSingletonByClassRef(const FUnrealObjectRef& SingletonClassRef)
+{
+	UClass* SingletonClass = Cast<UClass>(GetObjectFromUnrealObjectRef(SingletonClassRef));
+	if (SingletonClass != nullptr)
+	{
+		TArray<AActor*> FoundActors;
+		UGameplayStatics::GetAllActorsOfClass(NetDriver->World, SingletonClass, FoundActors);
```

Comment:
GetAllActorsOfClass looks like a heavy function - but as I understand it's called rarely, only when serializing/deserializing Singleton data?

Reply:
Yep, we use it as a last resort here to identify any active singleton actors that haven't registered themselves with the GSM yet.

Reply:
To add to that - this is only expected to get called at most a few times at the beginning, but it's worth keeping an eye on this being too slow and potentially optimizing in the future. It's more likely, however, that we'll rework singletons before that.